weblogic-azure-vm/arm-oraclelinux-wls-admin/src/main/arm/nestedtemplates/postDeploymentUAMIRolesTemplate.json

```diff
@@ -0,0 +1,114 @@
+{
+   "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+   "contentVersion": "1.0.0.0",
+   "parameters": {
+		"location": {
+		   	    "type": "string",
+			   "metadata": {
+			      "description": "Location for all resources."
+			   }
+		  },
+		  "_globalResourceNameSuffix": {
+		      "type": "string",
+		      "metadata": {
+		         "description": "A unique suffix that was specified during the deployment of the solution template."
+		    	}
+		   },
+		   "_artifactsLocation": {
+			     "type": "string",
+			     "metadata": {
+			        "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+			     }
+		   },
+		   "tagsByResource": {
+		      "type": "object",
+		      "defaultValue": {},
+		      "metadata": {
+		         "description": "${label.tagsLabel}"
+		      }
+		   },
+		   "roleAssignmentNameSeed": {
+		       "type": "string",
+		       "defaultValue": "[guid(subscription().id, parameters('_globalResourceNameSuffix'))]",
+		       "metadata": {
+		           "description": "A unique string used to generate the role assignment name. Defaults to a unique GUID based on the deployment context."
+		       }
+		   }		   
+	},
+	"variables": {
+		"const_roleDefinitionIdOfContributor": "b24988ac-6180-42a0-ab88-20f7382dd24c",
+		"name_postDeploymentScriptUserDefinedManagedIdentity": "[concat('post-deployment-user-defined-managed-identity', parameters('_globalResourceNameSuffix'))]",
+		"name_postDeploymentScriptRoleAssignment": "[concat('post-deployment-user-defined-role-assignment', parameters('_globalResourceNameSuffix'))]",
+		"name_postDeploymentScriptRoleAssignmentTemplate": "postDeploymentRoleAssignmentTemplate.json"
+	},
+	"resources": [
+		{		
+		    "type": "Microsoft.ManagedIdentity/userAssignedIdentities",
+		    "apiVersion": "${azure.apiVersionForIdentity}",
+		    "tags": "[parameters('tagsByResource')['${identifier.userAssignedIdentities}']]",
+		    "name": "[variables('name_postDeploymentScriptUserDefinedManagedIdentity')]",
+		    "location": "[parameters('location')]"
+		},
+		{
+			"type": "Microsoft.Resources/deployments",
+			"apiVersion": "${azure.apiVersionForDeployment}",
+			"name": "[variables('name_postDeploymentScriptRoleAssignment')]",

```

Comment:
You're trying to make a subscription scope deployment for role assignment, but are you missing specifying the `scope` property for the deployment? E.g., something like `"scope": "[concat('/subscriptions/', subscription().subscriptionId)]"`. You can use `az deployment sub list` or `az deployment sub show --name` to verify deployments at subscription scope.

Reply:
Update: it looks like property `scope` is not supported but `"subscriptionId": "[subscription().subscriptionId]"` does make the subscription scope deployment works. I found it from the ARM template that is trans-compiled from the Bicep of WebSphere single server offer, see the attachment for reference [twas-singleserver-mainTemplate.json](https://github.com/user-attachments/files/22438365/twas-singleserver-mainTemplate.json).